packages/mask/src/plugins/Wallet/SNSAdaptor/SelectNftContractDialog.tsx

```diff
@@ -162,6 +150,19 @@ export function SelectNftContractDialog(props: SelectNftContractDialogProps) {
     }, [id, setDialog])
     //#endregion
 
+    const { value: assets } = useNFTscanFindAssets(account, !open)
+
+    const erc721InDb = useERC721Tokens()
+    const allContractsInDb = unionBy(
+        erc721InDb.map((x) => x.contractDetailed),
+        'address',
+    ).map((x) => ({ contractDetailed: x, balance: undefined }))
+
+    const contractList =
+        chainId === ChainId.Mainnet && assets
+            ? unionBy([...assets, ...allContractsInDb], 'contractDetailed.address')
```

Comment:
You may need `formatEthereumAddress(address)` or `address.toLowerCase()` before comparing between them.

Reply:
Apply formatEthereumAddress() for nft scan api.